src/twisted/web/http_headers.py

```diff
@@ -168,7 +184,39 @@ def removeHeader(self, name: AnyStr) -> None:
 
         @return: L{None}
         """
-        self._rawHeaders.pop(self._encodeName(name), None)
+        self._rawHeaders.pop(_encodeName(name), None)
+
+    def _setRawHeadersFaster(self, name: _EncodedHeader, values: List[bytes]) -> None:
+        """
+        Like C{setRawHeaders}, but faster.
+
+        @param name: The name of the HTTP header to set the values for.
+
+        @param values: A list of bytes, each one being a header value of the
+            given name. These will be still be sanitized since otherwise it's
+            too easy for security bugs to allow unescaped whitespace through.
+        """
+        self._rawHeaders[name] = [_sanitizeLinearWhitespace(v) for v in values]
+
+    def _getRawHeaderLastFaster(self, name: _EncodedHeader) -> Optional[bytes]:
```

Comment:
Maybe we can have this as a public API

The argument can be bytes and users of this API are expected to call it with the canonical header name, already encoded.

```suggestion
    def getRawValue(self, name: bytes) -> Optional[bytes]:
```

this name should match the `Request.getHeader` API, in terms of use of singular